test/Algebra/Graph/Test/Generic.hs

```diff
@@ -1692,6 +1692,144 @@ testSimplify (prefix, API{..}) = do
+          bfsForestFrom vs empty                           == []
+
+    test "forest (bfsForestFrom [1]   $ edge 1 1)          == vertex 1" $
+          forest (bfsForestFrom [1]   $ edge 1 1)          == vertex 1
+
+    test "forest (bfsForestFrom [1]   $ edge 1 2)          == edge 1 2" $
+          forest (bfsForestFrom [1]   $ edge 1 2)          == edge 1 2
+
+    test "forest (bfsForestFrom [2]   $ edge 1 2)          == vertex 2" $
+          forest (bfsForestFrom [2]   $ edge 1 2)          == vertex 2
+
+    test "forest (bfsForestFrom [3]   $ edge 1 2)          == empty" $
+          forest (bfsForestFrom [3]   $ edge 1 2)          == empty
+
+    test "forest (bfsForestFrom [2,1] $ edge 1 2)          == vertices [1,2]" $
+          forest (bfsForestFrom [2,1] $ edge 1 2)          == vertices [1,2]
+
+    test "isSubgraphOf (forest $ bfsForestFrom vs x) x     == True" $ \vs x ->
+          isSubgraphOf (forest $ bfsForestFrom vs x) x     == True
+
+    test "bfsForestFrom (vertexList x) x                   == bfsForest x" $ \x ->
+          bfsForestFrom (vertexList x) x                   == bfsForest x
+
+    test "bfsForestFrom vs             (vertices vs)       == map (\\v -> Node v []) (nub vs)" $ \vs ->
+          bfsForestFrom vs             (vertices vs)       == map (\v -> Node v []) (nub vs)
+
+    test "bfsForestFrom []             x                   == []" $ \x ->
+          bfsForestFrom []             x                   == []
+
+    test "bfsForestFrom [1,4] $ 3 * (1 + 4) * (1 + 5)      == <correct result>" $
+          bfsForestFrom [1,4]  (3 * (1 + 4) * (1 + 5))     == [ Node { rootLabel = 1
+                                                                     , subForest = [ Node { rootLabel = 5
+                                                                                          , subForest = [] }]}
+                                                              , Node { rootLabel = 4
+                                                                     , subForest = [] }]
+          
+    test "bfsForestFrom [3] (circuit [1..5] + (circuit [5,4..1])) == <correct result>" $
+          bfsForestFrom [3] (circuit [1..5] + (circuit [5,4..1])) ==
+          [ Node { rootLabel = 3
+                 , subForest = [ Node { rootLabel = 2
+                                      , subForest = [ Node { rootLabel = 1
+                                                           , subForest = []}]}
+                               , Node { rootLabel = 4
+                                      , subForest = [ Node { rootLabel = 5
+                                                           , subForest = []}]}]}]
+          
+testBfs :: TestsuiteInt g -> IO ()
+testBfs (prefix, API{..}) = do
+    putStrLn $ "\n============ " ++ prefix ++ "bfs ============"
+    
+    test "bfs vs    $ empty                    == []" $ \vs ->
+          bfs vs      empty                    == []
```

Comment:
You have a lot of tests here, which I think would be useful examples. Could you add them to the docs?

Reply:
Done! Some of the lines are awkwardly long when showing the forest. Is it handled ok? Or should all the equals signs be aligned?

Reply:
I try to keep all `==` aligned both in Haddock and in the testsuite output, if possible. Happy to do the final polishing myself though.